### ISSUE TITLE
127.0.0.1 hardcoded in webagg backend server

### ISSUE THREAD

**Issue by thedch** [matplotlib/matplotlib#13055]
### Bug report

**Bug summary**

Hardcoded URL in webagg backend is misleading and sometimes wrong

**Code for reproduction**

```python
import matplotlib as mpl
mpl.use('WebAgg')
import matplotlib.pyplot as plt
import numpy as np

mpl.rcParams['webagg.open_in_browser'] = False
fig, plot = plt.subplots(nrows=1)
plot.imshow(np.random.rand(224,224,3))
plt.show()
```

**Actual outcome**

```
To view figure, visit http://127.0.0.1:8988
Press Ctrl+C to stop WebAgg server
```

**Expected outcome**

Hopefully something like 

```
To view figure, visit http://[insert my server IP here]:8988
Press Ctrl+C to stop WebAgg server
```

**Matplotlib version**

  * Operating system: Ubuntu 16.04
  * Matplotlib version: 2.0.2
  * Matplotlib backend (`print(matplotlib.get_backend())`): WebAgg
  * Python version: 3.6.1
  * Jupyter version (if applicable):
  * Other libraries: 

*** 

Basically, it'd be nice to have control over what IP gets printed there, instead of hard coding localhost -- https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/backends/backend_webagg.py#L324 -- maybe add another key to rcParams? (and leave localhost as the default?)

**Comment by anntzer** [comment]
Shouldn't we print `rcParams["webagg.address"]` here?  Not that I know anything about IPs...

**Comment by tacaswell** [comment]
To test that this works locally, try something like:

```python
import matplotlib as mpl
import netifaces
import numpy as np

mpl.use('WebAgg')
import matplotlib.pyplot as plt


mpl.rcParams['webagg.open_in_browser'] = False
mpl.rcParams['webagg.address'] = netifaces.ifaddresses('wlp4s0')[netifaces.AF_INET][0]['addr']
fig, plot = plt.subplots(nrows=1)
plot.imshow(np.random.rand(224, 224, 3))
plt.show()

```

I am hesitant to put this in the test suite as hard-coding any other ip address or interface name than `127.0.0.1` into the tests seems like it will be very brittle.
